sql-plugin/src/main/330+/scala/com/nvidia/spark/rapids/shims/v2/Spark33XShims.scala

```diff
@@ -153,6 +157,201 @@ trait Spark33XShims extends Spark33XFileOptionsShims {
+          TypeSig.all,
+          repeatingParamCheck = Some(RepeatingParamCheck("param",
+            (_gpuCommonTypes + TypeSig.DECIMAL_128 + TypeSig.ARRAY + TypeSig.STRUCT +
+                TypeSig.DAYTIME).nested(),
+            TypeSig.all))),
+        (a, conf, p, r) => new ExprMeta[Coalesce](a, conf, p, r) {
+          override def convertToGpu():
+          GpuExpression = GpuCoalesce(childExprs.map(_.convertToGpu()))
+        }),
+      GpuOverrides.expr[AttributeReference](
+        "References an input column",
+        ExprChecks.projectAndAst(
+          TypeSig.astTypes,
+          (TypeSig.commonCudfTypes + TypeSig.NULL + TypeSig.MAP + TypeSig.ARRAY +
+              TypeSig.STRUCT + TypeSig.DECIMAL_128 + TypeSig.DAYTIME).nested(),
+          TypeSig.all),
+        (att, conf, p, r) => new BaseExprMeta[AttributeReference](att, conf, p, r) {
+          // This is the only NOOP operator.  It goes away when things are bound
+          override def convertToGpu(): Expression = att
+
+          // There are so many of these that we don't need to print them out, unless it
+          // will not work on the GPU
+          override def print(append: StringBuilder, depth: Int, all: Boolean): Unit = {
+            if (!this.canThisBeReplaced || cannotRunOnGpuBecauseOfSparkPlan) {
+              super.print(append, depth, all)
+            }
+          }
+        }),
+      GpuOverrides.expr[TimeAdd](
+        "Adds interval to timestamp",
+        ExprChecks.binaryProject(TypeSig.TIMESTAMP, TypeSig.TIMESTAMP,
+          ("start", TypeSig.TIMESTAMP, TypeSig.TIMESTAMP),
+          // interval support DAYTIME column or CALENDAR literal
+          ("interval", TypeSig.DAYTIME + TypeSig.lit(TypeEnum.CALENDAR)
+              .withPsNote(TypeEnum.CALENDAR, "month intervals are not supported"),
+              TypeSig.DAYTIME + TypeSig.CALENDAR)),
+        (timeAdd, conf, p, r) => new BinaryExprMeta[TimeAdd](timeAdd, conf, p, r) {
+          override def tagExprForGpu(): Unit = {
+            GpuOverrides.extractLit(timeAdd.interval).foreach { lit =>
+              lit.dataType match {
+                case CalendarIntervalType =>
+                  val intvl = lit.value.asInstanceOf[CalendarInterval]
+                  if (intvl.months != 0) {
+                    willNotWorkOnGpu("interval months isn't supported")
+                  }
+                case _: DayTimeIntervalType => // Supported
+              }
+            }
+            checkTimeZoneId(timeAdd.timeZoneId)
+          }
+
+          override def convertToGpu(lhs: Expression, rhs: Expression): GpuExpression =
+            GpuTimeAdd(lhs, rhs)
+        })
+    ).map(r => (r.getClassFor.asSubclass(classOf[Expression]), r)).toMap
+    super.getExprs ++ map
+  }
+
+  // 330+ supports DAYTIME interval types
+  override def getExecs: Map[Class[_ <: SparkPlan], ExecRule[_ <: SparkPlan]] = {
+    val _gpuCommonTypes = TypeSig.commonCudfTypes + TypeSig.NULL + TypeSig.DECIMAL_64
```

Comment:
nit: everywhere that this is used the DECIMAL_64 is replaced with a DECIMAL_128.

Reply:
Updated.